src/Fhir.Anonymizer.Core.UnitTests/PartitionedExecution/FhirStreamConsumerTests.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Fhir.Anonymizer.Core.PartitionedExecution;
+using Xunit;
+
```

Comment:
may be helpful to add simulation of random I/O failure case in Reader and Consumer and check behaviors.

Reply:
Thanks, added 2 new unit test cases for ioexception from reader and consumer